lib/scope.js

```diff
@@ -909,36 +921,73 @@ module.exports = {
     // data for `users[0]`, there's no need to check the trigger var.
     // That way devs with simple cases can ignore the trigger var column.
 
-    // If field uses proxies and there has been more than one match
-    // we now use the trigger var to narrow down the matches.
-    if ( uses_proxies && matches.length > 1 ) {
-      // Multiple matches means this came from a story
-
-      // Feedback for devs - we need the trigger var but we don't have it (field data)
-      // TODO: Add docs and link to docs on how to add the trigger var
-      if ( field.trigger === undefined ) { await scope.addToReport(scope, { type: `warning`, value: `A field on this page uses an index variable or a generic object, but the page does not contain an element with the trigger variable's name. Tell the developer to add an element to the page with the id "trigger"` }); }
+    // Make sure the trigger var matches for a page that uses proxies. Only
+    // story tables need this disambiguation.
+    if ( trigger_might_be_needed && uses_proxies && matches.length > 0 ) {
```

Comment:
We are thinking that it might be easier to rename "matches" to "initial_matches" to avoid changing the meaning of the matches variable halfway through.